### PR TITLE
Promote plnsvc new ExternalSecret to prod

### PR DESCRIPTION
**Pull request by Roming22**
This is the first step in migrating the chains signing-secrets to Vault. This promotion will enable the validation of the ExternalSecret data before renaming it so it replaces the old one.

rh-pre-commit.version: 2.1.0
rh-pre-commit.check-secrets: ENABLED

### CODE REVIEW

**Comment by Roming22** [comment]
@hugares @gabemontero This PR does not modify development, therefore the CI is meaningless (except for the linter). Given the current issues on the CI, I propose to force merge the PR once approved.

File: components/pipeline-service/production/stone-prd-m01/resources/tekton-chains-signing-secret-path.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+- op: add
+  path: /spec/dataFrom/0/extract/key
+  value: production/pipeline-service/stone-stage-m01/chains-signing-secret
```
Comment (hugares) [inline components/pipeline-service/production/stone-prd-m01/resources/tekton-chains-signing-secret-path.yaml:4]: wrong key path? there is stage in the name of the secret

File: components/pipeline-service/production/stone-prd-rh01/resources/tekton-chains-signing-secret-path.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+- op: add
+  path: /spec/dataFrom/0/extract/key
+  value: production/pipeline-service/stone-stage-rh01/chains-signing-secret
```
Comment (hugares) [inline components/pipeline-service/production/stone-prd-rh01/resources/tekton-chains-signing-secret-path.yaml:4]: wrong key path? there is stage in the name of the secret

File: components/pipeline-service/production/stone-prod-p01/resources/tekton-chains-signing-secret-path.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+- op: add
+  path: /spec/dataFrom/0/extract/key
+  value: staging/pipeline-service/stone-stage-p01/chains-signing-secret
```
Comment (hugares) [inline components/pipeline-service/production/stone-prod-p01/resources/tekton-chains-signing-secret-path.yaml:4]: wrong key path? there is stage in the name of the secret

Comment (Roming22) [reply components/pipeline-service/production/stone-prod-p01/resources/tekton-chains-signing-secret-path.yaml]: Yup, I forgot to save the file :disappointed: 
Fixed.

File: components/pipeline-service/production/stone-prd-m01/resources/tekton-chains-signing-secret-path.yaml (hunk 1 above)
Comment (hugares) [inline components/pipeline-service/production/stone-prd-m01/resources/tekton-chains-signing-secret-path.yaml:4]:
```suggestion
  value: production/pipeline-service/stone-prod-m01/chains-signing-secret
```

File: components/pipeline-service/production/stone-prd-rh01/resources/tekton-chains-signing-secret-path.yaml (hunk 2 above)
Comment (hugares) [inline components/pipeline-service/production/stone-prd-rh01/resources/tekton-chains-signing-secret-path.yaml:4]:
```suggestion
  value: production/pipeline-service/stone-prod-rh01/chains-signing-secret
```

File: components/pipeline-service/production/stone-prod-p01/resources/tekton-chains-signing-secret-path.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+- op: add
+  path: /spec/dataFrom/0/extract/key
+  value: production/pipeline-service/stone-stage-p01/chains-signing-secret
```
Comment (hugares) [inline components/pipeline-service/production/stone-prod-p01/resources/tekton-chains-signing-secret-path.yaml:4]:
```suggestion
  value: production/pipeline-service/stone-prod-p01/chains-signing-secret
```

**Review by hugares — COMMENTED** [review]
You also need to kustomize build after fixing issues and commit deploy files

File: components/pipeline-service/production/stone-prd-m01/deploy.yaml [hunk 5]
```diff
@@ -1712,6 +1712,27 @@ spec:
...
+metadata:
+  annotations:
+    argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
+    argocd.argoproj.io/sync-wave: "-1"
+  name: tekton-chains-signing-secret
+  namespace: openshift-pipelines
+spec:
+  dataFrom:
+  - extract:
+      key: production/pipeline-service/stone-stage-m01/chains-signing-secret
```
Comment (gabemontero) [inline components/pipeline-service/production/stone-prd-m01/deploy.yaml:1724]: the key still says `stone-stage-m01` @Roming22

File: components/pipeline-service/production/stone-prd-rh01/deploy.yaml [hunk 6]
```diff
@@ -1712,6 +1712,27 @@ spec:
...
+metadata:
+  annotations:
+    argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
+    argocd.argoproj.io/sync-wave: "-1"
+  name: tekton-chains-signing-secret
+  namespace: openshift-pipelines
+spec:
+  dataFrom:
+  - extract:
+      key: production/pipeline-service/stone-stage-rh01/chains-signing-secret
```
Comment (gabemontero) [inline components/pipeline-service/production/stone-prd-rh01/deploy.yaml:1724]: still says stage here too

File: components/pipeline-service/production/stone-prod-p01/deploy.yaml [hunk 7]
```diff
@@ -1712,6 +1712,27 @@ spec:
...
+metadata:
+  annotations:
+    argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
+    argocd.argoproj.io/sync-wave: "-1"
+  name: tekton-chains-signing-secret
+  namespace: openshift-pipelines
+spec:
+  dataFrom:
+  - extract:
+      key: production/pipeline-service/stone-stage-p01/chains-signing-secret
```
Comment (gabemontero) [inline components/pipeline-service/production/stone-prod-p01/deploy.yaml:1724]: and still says stage here

File: components/pipeline-service/production/stone-prd-m01/resources/tekton-chains-signing-secret-path.yaml (hunk 1 above)
Comment (Roming22) [reply components/pipeline-service/production/stone-prd-m01/resources/tekton-chains-signing-secret-path.yaml]: Fixed

File: components/pipeline-service/production/stone-prd-rh01/resources/tekton-chains-signing-secret-path.yaml (hunk 2 above)
Comment (Roming22) [reply components/pipeline-service/production/stone-prd-rh01/resources/tekton-chains-signing-secret-path.yaml]: Fixed

File: components/pipeline-service/production/stone-prd-m01/resources/tekton-chains-signing-secret-path.yaml (hunk 1 above)
Comment (Roming22) [reply components/pipeline-service/production/stone-prd-m01/resources/tekton-chains-signing-secret-path.yaml]: Fixed

File: components/pipeline-service/production/stone-prd-rh01/resources/tekton-chains-signing-secret-path.yaml (hunk 2 above)
Comment (Roming22) [reply components/pipeline-service/production/stone-prd-rh01/resources/tekton-chains-signing-secret-path.yaml]: Fixed

File: components/pipeline-service/production/stone-prod-p01/resources/tekton-chains-signing-secret-path.yaml (hunk 4 above)
Comment (Roming22) [reply components/pipeline-service/production/stone-prod-p01/resources/tekton-chains-signing-secret-path.yaml]: Fixed

File: components/pipeline-service/production/stone-prd-m01/deploy.yaml (hunk 5 above)
Comment (Roming22) [reply components/pipeline-service/production/stone-prd-m01/deploy.yaml]: Fixed

File: components/pipeline-service/production/stone-prd-rh01/deploy.yaml (hunk 6 above)
Comment (Roming22) [reply components/pipeline-service/production/stone-prd-rh01/deploy.yaml]: Fixed

File: components/pipeline-service/production/stone-prod-p01/deploy.yaml (hunk 7 above)
Comment (Roming22) [reply components/pipeline-service/production/stone-prod-p01/deploy.yaml]: Fixed

File: components/pipeline-service/staging/base/kustomization.yaml [hunk 8]
```diff
@@ -9,7 +9,6 @@ commonAnnotations:
 
 resources:
   - https://github.com/openshift-pipelines/pipeline-service.git/operator/gitops/argocd/pipeline-service?ref=2be0e3a49809ba66bf64625d01833de90c457094
-  - chains-signing-secrets.yaml
```
Comment (gabemontero) [inline components/pipeline-service/staging/base/kustomization.yaml:12]: this was already missing in https://github.com/redhat-appstudio/infra-deployments/blob/main/components/pipeline-service/production/base/kustomization.yaml 

is that expected @Roming22 ?

Comment (Roming22) [reply components/pipeline-service/staging/base/kustomization.yaml:12]: Yes. As part of this PR I moved the config from `staging/base` to `base`.

Comment (Roming22) [reply components/pipeline-service/staging/base/kustomization.yaml:12]: Locally I also regenerated the `deploy.yaml` for staging to check that they were not impacted.

**Review by hugares — APPROVED** [review]
/lgtm

File: components/pipeline-service/staging/base/kustomization.yaml (hunk 8 above)
Comment (gabemontero) [reply components/pipeline-service/staging/base/kustomization.yaml:12]: sgtm thanks

**Review by gabemontero — APPROVED** [review]
/lgtm

fingers crossed / good luck

**Pull request merged**